### PR TITLE
from grpc block ProduceBlock mapping - Performance+Typing

### PR DESCRIPTION
**Pull request by grooviegermanikus**
- remove calls to Pubkey::default()
- use Signature for .signature instead of String
- use Hash for recent_blockhash instead of String
- refactor reading of compute_budget instructions
- move message serialization

### CODE REVIEW

**Comment by grooviegermanikus** [comment]
FYI deployed tag experimental/performance-from_grpc_block_update-01

**Review by godmodegalactus — DISMISSED** [review]

File: cluster-endpoints/src/grpc_subscription.rs [hunk 1]
```diff
@@ -272,6 +203,63 @@ pub fn from_grpc_block_update(
     ProducedBlock::new(inner, commitment_config)
 }
 
+fn map_compute_budget_instructions(message: &VersionedMessage) -> (Option<u32>, Option<u64>) {
+    let cu_requested_cell: OnceCell<Option<u32>> = OnceCell::new();
```
Comment (godmodegalactus) [inline cluster-endpoints/src/grpc_subscription.rs:207]: Onecell seems little useless with option, eigher should go with OnceCell or Option.

File: cluster-endpoints/src/grpc_subscription.rs [hunk 2]
```diff
@@ -272,6 +203,63 @@ pub fn from_grpc_block_update(
...
+                        .expect("legacy price+fees must be set only once");
+                }
+                _ => {
+                    trace!("skip compute budget instruction");
+                }
+            }
+        }
+    }
+
+    let legacy = legacy_cell.into_inner();
```
Comment (godmodegalactus) [inline cluster-endpoints/src/grpc_subscription.rs:253]: Can remove legacy cell and do everything into prioritization fees.

**Review by godmodegalactus — CHANGES_REQUESTED** [review]

**Comment by grooviegermanikus** [comment]
deploing experimental/performance-from_grpc_block_update-02

**Review by godmodegalactus — APPROVED** [review]

**Pull request merged**